packages/rocketchat-ui-flextab/flex-tab/tabs/uploadedFilesList.coffee

```diff
@@ -23,11 +23,40 @@ Template.uploadedFilesList.helpers
 	escapedName: ->
 		return s.escapeHTML @name
 
+	canDelete: ->
+		msg = ChatMessage.findOne { file: { _id: @_id } }
+		if msg
+			return RocketChat.authz.hasAtLeastOnePermission('delete-message', msg.rid) or RocketChat.settings.get('Message_AllowDeleting') and msg.u?._id is Meteor.userId()
+
 Template.uploadedFilesList.events
 	'click .room-file-item': (e, t) ->
 		if $(e.currentTarget).siblings('.icon-picture').length
 			e.preventDefault()
 
+	'click .icon-trash': (e, t) ->
+		self = this
+		swal {
+			title: 'Are you sure?'
+			text: 'You will not be able to recover this file.'
```

Comment:
This text should be translatable, use `TAPi18n.__('You_will_not_be_able_to_recover_this_file)` and add `You_will_not_be_able_to_recover_this_file` to `i18n/en.i18n.json`